saleor/graphql/payment/mutations.py

```diff
@@ -1097,21 +1041,24 @@ def update_amounts_for_order(
     def update_transaction(cls, instance, transaction_data):
         if instance.order_id:
             cls.update_amounts_for_order(instance, instance.order_id, transaction_data)
-        transaction_data["psp_reference"] = transaction_data.get(
+        psp_reference = transaction_data.get(
             "psp_reference", transaction_data.pop("reference", None)
         )
+        if psp_reference and instance.psp_reference != psp_reference:
+            if payment_models.TransactionItem.objects.filter(
+                psp_reference=psp_reference
+            ).exists():
+                raise ValidationError(
+                    {
+                        "transaction": ValidationError(
```

Comment:
Didn't we want to drop `unique` on `pspReference`? Or is not the case here?

Reply:
Thanks for catching this. I think that we will have a validation on the pair: order+transaction's pspReference. I have this on my todo list (on main feature branch). There is still some things to clarify, and I didn't want to extend this PR more.
![image](https://user-images.githubusercontent.com/23282005/208910952-e69b537a-ec5a-4c5a-9d92-d36657aa71e0.png)